### ISSUE TITLE
Are message ids in a chat guaranteed to monotonously increase in chronological order?

### ISSUE THREAD

**Issue by ars3niy** [tdlib/td#1378]
They seem to do that. Is that something that's guaranteed to be true forever?

**Comment by levlam** [comment]
No. Especially, this can be false in secret chats.

But messages must be sorted by their identifier regardless of their date. Most of the time the messages will be sorted by date also.

**Comment by ars3niy** [comment]
OK let me rephrase. If I stay online forever and watch updateNewMessage arrive, is it true that message id will always increase in order of arrival?

If I go offline, log in later and receive a bunch of message at login (that were sent while I was offline), is it true that their message id will be bigger than that of the last message I saw before logout?

**Comment by levlam** [comment]
The message ID in `updateNewMessage` is guaranteed to always increase within one TDLib launch if the current user is not a bot.

If the user is not a bot, message database is enabled, the previous launch was generously closed using `close` and the database isn't broken, then it will be true also between restarts.

Relogin is roughly equivalent to full database deletion, so there are cases in which you can receive again `updateNewMessage` for messages with lower message IDs.

**Comment by ars3niy** [comment]
I see, thanks!
